src/cljs/main/broadfcui/persistence.cljs

```diff
@@ -19,10 +19,28 @@
 (defn try-restore [{:keys [key initial validator]}]
   (let [saved-state (some-> key generate-persistence-key utils/local-storage-read cljs.reader/read-string)]
     (if (and saved-state
-             (or (not validator) (some-> saved-state validator)))
+             (or (not validator) (validator saved-state)))
       saved-state
       (initial))))
 
 (defn delete [key]
   (utils/local-storage-remove
    (generate-persistence-key key)))
+
+
+(defn with-state-persistence [{:keys [key version initial except only]} defined-methods]
+  (let [get-initial-state
+        (fn [data]
+          (merge
+           (try-restore
+            {:key key
+             :initial (fn [] (merge initial (when version {:v version})))
+             :validator (when version (comp (partial = version) :v))})
+           (when-let [defined (:get-initial-state defined-methods)]
+             (defined data))))
+        component-did-update
+        (fn [{:keys [state] :as data}]
+          (save {:key key :state state :except except :only only})
+          (when-let [defined (:component-did-update defined-methods)]
+            (defined data)))]
+    (assoc defined-methods :get-initial-state get-initial-state :component-did-update component-did-update)))
```

Comment:
is it worthwhile/possible for callers to specify their own validator? I'm wondering if there is a use case for certain components to validate e.g. "as long as localstate's version is greater than or equal to what the component specifies", or "as long as localstate's version is between 3 and 6"